CHANGELOG.md

```diff
@@ -8,6 +8,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 - Support for HTTP/2 (server-side) ([#3847](https://github.com/opensearch-project/OpenSearch/pull/3847))
 - Add getter for path field in NestedQueryBuilder ([#4636](https://github.com/opensearch-project/OpenSearch/pull/4636))
 - Allow mmap to use new JDK-19 preview APIs in Apache Lucene 9.4+ ([#5151](https://github.com/opensearch-project/OpenSearch/pull/5151))
+- Removed NOT_FOUND for empty results. ([#6544](https://github.com/opensearch-project/OpenSearch/pull/6544))
```

Comment:
If this needs to be backported to `2.x` branch, we can directly add this under `[Unrelease 2.x]` section.

Reply:
I am not sure about it as this is my first commit. If this needs to be done, let me know. I will add it and push the commit.

Reply:
I guess it makes sense to backport it 2.x branch since this is a bug. Pushing the commit.